src/index.ts

```diff
@@ -178,16 +164,22 @@ async function setUpAutoCache() {
   // Start the server. Once it is ready, it will notify us via the notification server.
   const daemon = spawn(daemonBin, daemonCliFlags, opts);
 
-  const pidFile = path.join(daemonDir, 'daemon.pid');
+  const pidFile = path.join(daemonDir, "daemon.pid");
   await fs.writeFile(pidFile, `${daemon.pid}`);
 
   core.info("Waiting for magic-nix-cache to start...");
 
   await new Promise<void>((resolve, reject) => {
-    notifyPromise.then((value) => {
-      resolve();
-    });
-    daemon.on('exit', async (code, signal) => {
+    notifyPromise
+      // eslint-disable-next-line github/no-then
+      .then((_value) => {
+        resolve();
+      })
+      // eslint-disable-next-line github/no-then
+      .catch((err) => {
+        reject(new Error(`error in notifyPromise: ${err}`));
+      });
+    daemon.on("exit", async (code, signal) => {
```

Comment:
hm, I'm not good at TS, it's not obvious to me what this does. the outer Promise is awaited, and your're registering callback that calls that Promise's reject? but not until the daemon exits? My brain can't figure out what the semantics of this are.

Reply:
Okay, I guess it is intentionally blocking to wait for MNC daemon to ping us on notifyPort.